### PR TITLE
Add an 'on-start.md' to note event registration order.

### PR DESCRIPTION
**Pull request by ganicke**
Related to https://github.com/Microsoft/pxt/pull/2972.

Add a note for event registration order for **forever** and **runInBackground**.

### CODE REVIEW

File: docs/blocks/on-start.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# @extends
+
+## #eventorder
+
+## ~hint
+**Who goes first?**
+
+All events are registered before the **onStart** block runs. If your program has a **[forever](/reference/loops/forever)** loop or a **[runInBackground](/reference/control/run-in-background)** block, these will not start until after the code in **onStart** begins.
+
```
Comment (pelikhan) [inline docs/blocks/on-start.md]: runs. Events will start running as soon as a pause (or any other async operation) occurs in your code.

Comment (pelikhan) [reply docs/blocks/on-start.md]: onStart **ends**.

Comment (ganicke) [reply docs/blocks/on-start.md]: **forever** and **runInBackground** are blocked until any **pause()** in **onStart**? Or, they just don't start until **onStart** is done?

**Comment by pelikhan** [comment]
Those 2 don't start until onStart is done. The events start as soon as pause happens

**Comment by ganicke** [comment]
How about now?

image

**Comment by pelikhan** [comment]
Need to wait for release

**Pull request merged**